pkg/commands/edit/set/setimage.go

```diff
@@ -197,5 +197,12 @@ func parseOverwrite(arg string) (overwrite, error) {
 			tag:  t[2],
 		}, nil
 	}
+
+	// match <image>
+	if len(arg) > 0 && overwriteImage {
```

Comment:
I don't think we need the boolean `overwriteImage`. When there is no `:` or `@` in `arg`, it can be treated as the new image name.

Reply:
Hi @Liujingfang1, it is actually needed,  if removed this command would be treated as valid: `kustomize edit set image foo`, which makes no sence, note that the `parseOverwrite` function is used in two contexts.  Test `TestSetImage/error:_invalid_args[bad_args]` also breaks, which guards agains this invalid command.